### ISSUE TITLE
dataExporter: lazy dataTable rows attribute not restored after exportation

### ISSUE THREAD

**Issue by cmaulini** [primefaces/primefaces#2958]
## 1) Environment
- PrimeFaces version: 6.1.8
- Does it work on the newest released PrimeFaces version? Version? No.
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source) No.
- Application server + version: Apache Tomcat 8.5.23
- Affected browsers: All.

## 2) Expected behavior

When a lazy dataTable is exported using `<p:dataExporter>`, once exportation is done, the `org.primefaces.component.export.Exporter.exportAll(FacesContext, DataTable, Object)` method should restore the `rows` attribute. Otherwise,the 2th `LazyDataModel#load(..)` is called with bad `pageSize` (`pageSize == rowCount`).

## 3) Actual behavior

The `<p:dataExporter>` doesn't restore `rows` attribute. After export, the reloading of the dataTable load too much data!

## 4) Steps to reproduce

1) create a lazy dataTable
2) run an export
3) the 2th `LazyDataModel#load(..)` has bad `pageSize` 

## 5) Sample XHTML

```
<h:form>
	<p:dataTable id="lazyTable" var="obj"
		value="#{testDataExportView.lazyDataModel}" paginator="true" rows="10"
		lazy="true"
		paginatorTemplate="{CurrentPageReport}  {FirstPageLink} {PreviousPageLink} {PageLinks} {NextPageLink} {LastPageLink}">

		<p:column>
			<f:facet name="header">
				<h:outputText value="ID" />
			</f:facet>
			<h:outputText value="#{obj.id}" />
		</p:column>

		<p:column>
			<f:facet name="header">
				<h:outputText value="Str" />
			</f:facet>
			<h:outputText value="#{obj.str}" />
		</p:column>
	</p:dataTable>
	<p:commandButton value="Export" ajax="false">
		<p:dataExporter type="csv" target="lazyTable" fileName="objs"
			pageOnly="false" />
	</p:commandButton>
</h:form>
```

## 6) Sample bean

```
@ManagedBean
public class TestDataExportView implements Serializable {

	private final LazyObjDataModel mLazyDataModel = new LazyObjDataModel();

	public LazyObjDataModel getLazyDataModel() {
		return mLazyDataModel;
	}

	public static class LazyObjDataModel extends LazyDataModel<LazyObj> {
		@Override
		public List<LazyObj> load(final int first, final int pageSize,
				final String sortField, final SortOrder sortOrder,
				final Map<String, Object> filters) {
			setRowCount(5000);
			final List<LazyObj> result = new ArrayList<>(pageSize);
			final int last = first + pageSize;
			for (int i = first; i < last; i++) {
				result.add(new LazyObj(i, String.format("The %dth object", i)));
			}
			return result;
		}
	}

	public static class LazyObj implements Serializable {
		private final int mId;
		private final String mStr;

		public LazyObj(final int id, final String str) {
			mId = id;
			mStr = str;
		}

		public int getId() {
			return mId;
		}

		public String getStr() {
			return mStr;
		}
	}

}
```
## 7) Proposal

Add the `table.setRows(rows);` line in method `org.primefaces.component.export.Exporter.exportAll(FacesContext, DataTable, Object)` :

```
    protected void exportAll(FacesContext context, DataTable table, Object document) {
        int first = table.getFirst();
    	int rowCount = table.getRowCount();
        int rows = table.getRows();
        boolean lazy = table.isLazy();
        
        if(lazy) {
        	 if(rowCount > 0)
             {
                table.setFirst(0);
                table.setRows(rowCount);
                table.clearLazyCache();
                table.loadLazyData();
             }

            for(int rowIndex = 0; rowIndex < rowCount; rowIndex++) {
                exportRow(table, document, rowIndex);
            }
     
            //restore
            table.setFirst(first);
            table.setRowIndex(-1);
            table.setRows(rows); // RESTORE PROPERLY ROWS ATTRIBUTE
            table.clearLazyCache();
            table.loadLazyData();
        } 
        else {
            for(int rowIndex = 0; rowIndex < rowCount; rowIndex++) {                
                exportRow(table, document, rowIndex);
            }
            
            //restore
            table.setFirst(first);
        }
    }
```

**Comment by tandraschko** [comment]
Already done in trunk, just search in the issue tracker next time ;) it's up to @mertsincan to port it to elite.
